### PR TITLE
Preserves quotes around paths with spaces when invoking group member scheduled task

### PR DESCRIPTION
**Pull request by lorengordon**

### CODE REVIEW

**Comment by lorengordon** [comment]
This basically changes the command from this:

```
C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive -ExecutionPolicy Bypass -Command "{{ InvokeScript }}" -Path "{{ RegisterTaskScript }}" -TaskName "{{ TaskName }}" -Members '{{ Admins }}' -DomainNetBiosName {{ DomainName }}
```

to this:


```
C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive -ExecutionPolicy Bypass -Command "& '{{ InvokeScript }}' -Path '{{ RegisterTaskScript }}' -TaskName '{{ TaskName }}' -Members '{{ Admins }}' -DomainNetBiosName {{ DomainName }}"
```

**Comment by lorengordon** [comment]
Completed testing and confirmed this is working with both salt 3003 and 3004, across ws2012r2, ws2016, and ws2019!

File: join-domain/windows/files/Register-RunOnceStartupTask.ps1 [hunk 1]
```diff
@@ -24,14 +24,14 @@ $PowerShellArgs = @(
...
     "-Path"
-    "`"${RunOnceScript}`""
+    "'${RunOnceScript}'"
     "-TaskName"
-    "`"${TaskName}`""
+    "'${TaskName}'"
 )
 
-$TaskArgs = ${PowerShellArgs} + ${WrapperArgs} + ${RemainingArgs}
+$TaskArgs = ${PowerShellArgs} + @("`"&") + ${WrapperArgs} + ${RemainingArgs} + @("`"")
```
Comment (eemperor) [inline join-domain/windows/files/Register-RunOnceStartupTask.ps1:34]: Wow.  That's some wild mix of sub-expression and escape usage.

**Review by eemperor — APPROVED** [review]

File: join-domain/windows/files/Register-RunOnceStartupTask.ps1 (hunk 1 above)
Comment (lorengordon) [reply join-domain/windows/files/Register-RunOnceStartupTask.ps1:34]: Thanks. I hate it, and yet couldn't come up with a better approach. For the cmd shell, it _must_ be `"` in the outer quotes, and `'` for the inner quotes, it is very particular!

**Pull request merged**